firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateApkClient.java

```diff
@@ -57,51 +60,51 @@ public void updateApk(
       @NonNull String downloadUrl,
       @NonNull Activity currentActivity) {
 
-    downloadApk(downloadUrl, updateTask)
+    this.cachedUpdateTask = updateTask;
+    downloadApk(downloadUrl)
         .addOnSuccessListener(
             downloadExecutor,
             file ->
                 install(file.getPath(), currentActivity)
                     .addOnSuccessListener(
```

Comment:
Does the onSuccess listener ever run? Or does the successful install kill the process before it executes?

Reply:
ha that's a great point. Its code that would never run. but might be worth leaving it in there anyway? maybe?

Reply:
I guess it feels weird that we have an UpdateStatus in our API (`INSTALLED`) that we can never actually return.

Maybe we should remove that state?

Reply:
Tested it out and it kills the process, so it seems like it might be impossible to set the Install terminals state